tests/utils.rs

```diff
@@ -14,12 +14,47 @@ use std::path::PathBuf;
 use test_utils::{trim_margin, TempFile};
 use yaml_rust::YamlLoader;
 
-fn compare_results(result: (ExitCode, String), expected: Result<(), &str>) {
-    let expected_output = match expected {
-        Err(expected_output) => (ExitCode(1), expected_output.to_string()),
-        Ok(()) => (ExitCode(0), "All tests passed.\n".to_string()),
-    };
-    assert_eq!(result, expected_output);
+#[derive(Debug, PartialEq)]
+pub struct Expect {
```

Comment:
"Expect" feels a bit generic for what this does. the verbose description i can think of is "ScriptRunEndState", but that's probably too wordy. any ideas?

Reply:
My preference is to put descriptiveness in the method names, as opposed to the name of the type itself. To that end I changed the names of the `Expect` methods:

- `ok` → `tests_pass`
- `err` → `error_message`
- `stdout` → `with_stdout` (per your suggestion)
- `stderr` → `with_stderr`

Although this type is specialized for our domain, I think it would be nice for it to read like a sentence as expectations in some test frameworks do. For example, `Expect::error_message("failure").with_stdout("test")`